src/app/api/v1/admin/feature-flags/route.ts

```diff
@@ -0,0 +1,90 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { NextRequest, NextResponse } from "next/server";
+import { getToken } from "next-auth/jwt";
+import {
+  getAllFeatureFlags,
+  addFeatureFlag,
+  enableFeatureFlagByName,
+  disableFeatureFlagByName,
+  updateAllowList,
+  updateWaitList,
+} from "../../../../../db/tables/featureFlags";
+
+import type { FeatureFlag } from "../../../../../db/tables/featureFlags";
+import { isAdmin } from "../../../utils/auth";
+import appConstants from "../../../../../appConstants";
+
+export async function GET(req: NextRequest) {
+  const token = await getToken({ req });
+  if (typeof token?.email === "string" && isAdmin(token?.email)) {
+    // Signed in
```

Comment:
Same goes for the other request handlers, but I like the pattern of favouring early returns over `if...else` clauses, to make it easy to inspect required preconditions, as well as to reduce indentation:

```suggestion
  if (typeof token?.email !== "string" || !isAdmin(token?.email)) {
    return NextResponse.redirect(appConstants.SERVER_URL, 301);
  }
  // <contents of if{} block>
```